pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -123,9 +124,12 @@ def advance(
         self.trainer.logger_connector.update_eval_step_metrics()
 
         # track epoch level outputs
-        self.outputs = self._track_output_for_epoch_end(self.outputs, output)
+        if self._should_track_batch_outputs_for_epoch_end():
+            output = recursive_detach(output, to_cpu=self.trainer.move_metrics_to_cpu)
```

Comment:
before this didn't cause an error because the `_track_output_for_epoch_end` function that got removed in this PR seems to not call `recursive_detach` for `defaultdict`

Reply:
Ultimately this error is caused by Line 105 in the `apply_to_collection` function in `utilities/apply_func.py`
https://github.com/PyTorchLightning/pytorch-lightning/blob/1686aab5506ed6f4fee8683ce6cca711e62b5ef0/pytorch_lightning/utilities/apply_func.py#L105

as basically this line is doing 
`defaultdict(OrderedDict([...]))`
which would cause the same error:
```bash
TypeError: first argument must be callable or None
```

Reply:
I'll make a separate issue out of this so that others can search the same.

Reply:
Issue left at #10308